internal/supervisor/import.go

```diff
@@ -88,16 +81,52 @@ func (s *ImportSupervisor) supervise() error {
 			continue
 		}
 
-		// if the State is stable and the Installation is slightly old, we
-		// can safely assume that an import happened and was completed
-		if installation.State == "stable" {
-			i.CompleteAt = time.Now().Unix() / 1000
+		if startedImportIsComplete(installation, i) {
+			i.CompleteAt = model.Timestamp()
 			err := s.store.UpdateImport(i)
 			if err != nil {
-				s.logger.WithError(err).Warnf("Import %s was done but couldn't be marked as such", i.ID)
+				s.logger.WithError(err).Warnf("Import %s was complete but couldn't be marked as such", i.ID)
 			}
+
+			key := fmt.Sprintf("%s.zip", i.TranslationID)
+			cfg, err := config.LoadDefaultConfig(context.TODO())
+			if err != nil {
+				s.logger.WithError(err).Warnf("Failed to clean up s3://%s/%s", s.bucket, key)
+				return err
+			}
+
+			ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
+			defer cancelFunc()
+
+			client := s3.NewFromConfig(cfg)
+			_, err = client.DeleteObject(ctx, &s3.DeleteObjectInput{
+				Bucket: &s.bucket,
+				Key:    &key,
+			})
+
+			return err
 		}
 	}
 
 	return nil
 }
+
+// startedImportIsComplete returns true if an Import with a nonzero
+// StartAt value has been completed, and false otherwise.
+func startedImportIsComplete(installation *cloud.InstallationDTO, i *model.Import) bool {
+	switch {
+	case
+		// go ahead and mark Imports against Deleted Installations as
+		// complete
+		installation.State == cloud.InstallationStateDeleted:
+	case
+		// if the State of the Installation is stable and the Import was
+		// started at least a little while ago, we can safely assume that
+		// an import happened and was completed
+		installation.State == cloud.InstallationStateStable &&
+			i.StartAt < (model.Timestamp()-(time.Minute.Nanoseconds()/1000)):
```

Comment:
That edge case is addressed but it's a bit hard to reason about so I'll try to explain as best as I can.

When the lock fails, the Provisioner marks the Import complete and reports the error in the `Error` field, so the race condition doesn't occur. The Import supervisor comes back the next time it wakes up and finds the Import marked complete (with error!) and just moves on. 

The only way this race condition can occur is if the Provisioner succeeds in claiming the Import, but then fails to lock the Installation _and_ fails to talk back about the failure to lock to the AWAT -- in which case, the logic here will mark the Import done when it hasn't been performed, and it'll require manual reconciliant -- the Import has to be restarted manually. But that's the only downside.

To be clear: if the lock fails, the Import will require manual reconciliation _anyway_. In the original design, this is where the Autohealer would be running on another thread to eventually come along, look at Imports that failed with error, and possibly be able to automatically reconcile the state at a later time.

Reply:
>If I am not mistaken Provisioner will never come back to this import as GetTranslationReadyToImport takes only imports with StartAt == 0. Installation was Stable the whole time tho, so after a minute pass, this will set it as complete even tho nothing has happened on the Provisioner side.

Yeah, this is actually intended, as I only wanted to stop a catastrophic or eternally-repeating error here, or a race condition when no lock is involved (as in the previous version of the code).

I have been reluctant to add an Installation state in the Provisioner of `import-complete` because conceptually it is the same as `stable` and because I was concerned about edge cases _there,_ but it makes this logic much simpler (basically just have the AWAT poll until the Installation state becomes `import-complete`, then have the AWAT issue a command to move the Installation to `stable` and mark the Import `import-complete` in its database) and the more I struggle with the myriad of possible corner cases in trying to navigate with logic _without_ that extra state, the more I'm thinking it's an error to try to continue without an Installation state of `import-complete`... let me just.. spike on that... hold tight, PR incoming

Reply:
This is addressed with commit 0a81517 and PR https://github.com/mattermost/mattermost-cloud/pull/498